src/test/sock_tests.cpp

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) 2021-2021 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef WIN32
+
+#include <test/util/setup_common.h>
+#include <util/sock.h>
+
+#include <boost/test/unit_test.hpp>
+
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <thread>
+
+using namespace std::chrono_literals;
+
+BOOST_FIXTURE_TEST_SUITE(sock_tests, BasicTestingSetup)
+
+static bool SocketIsClosed(const SOCKET& s)
+{
+    // Notice that if another thread is running and creates its own socket after `s` has been
+    // closed, it may get assigned the same file descriptor number. In this case our test will
+    // wronly pretend that the socket is not closed.
+    int type;
+    socklen_t len = sizeof(type);
+    return getsockopt(s, SOL_SOCKET, SO_TYPE, &type, &len) == -1 &&
+           (errno == EBADF || errno == ENOTSOCK);
+}
+
+static SOCKET CreateSocket()
+{
+    const int s = socket(AF_INET, SOCK_STREAM, IPPROTO_TCP);
+    BOOST_REQUIRE(s != -1);
```

Comment:
It could make sense to use `SOCKET_ERROR` here.

Reply:
Done, thanks!